src/contrib/cluster/Akka.Cluster.Sharding/Shard.cs

```diff
@@ -658,12 +658,15 @@ bool AwaitingLease(object message)
                 case Shard.LeaseLost ll:
                     shard.HandleLeaseLost(ll);
                     return true;
-
-                case var _ when shard.ExtractEntityId(message).HasValue:
-                    shard.DeliverMessage(message, shard.Context.Sender);
+                default:
+                {
+                    var extracted = shard.ExtractEntityId(message);
+                    if (!extracted.HasValue) return false;
+                    
+                    shard.DeliverTo(extracted.Value.Item1, message, extracted.Value.Item2, shard.Context.Sender);
```

Comment:
Is there a reason this got changed to `shard.DeliverTo` when the old code did `shard.DeliverMessage`?